src/api/extendInstallOptions.ts

```diff
@@ -165,6 +165,9 @@ export default async (
   if (extendedOpts.lock === false && !extendedOpts.reinstallForFlatten) {
     logger.warn('using --no-lock I sure hope you know what you are doing')
   }
+  if (extendedOpts.shamefullyFlatten && !extendedOpts.reinstallForFlatten) {
+    logger.warn('using --shamefully-flatten is discouraged, you should declare all of your dependencies in package.json')
```

Comment:
it should be enough to have it only here. It will be printed on install and update. I don't think we need it on other commands